krabs/krabs/filtering/event_filter.hpp

```diff
@@ -107,8 +116,13 @@ namespace krabs {
     : predicate_(predicate)
     {}
 
-	inline event_filter::event_filter(unsigned short event_id, filter_predicate predicate/*=nullptr*/)
-    : provider_filter_event_id_(event_id),
+    inline event_filter::event_filter(std::vector<unsigned short> event_ids, filter_predicate predicate/*=nullptr*/)
+    : provider_filter_event_ids_(event_ids.begin(), event_ids.end()),
+      predicate_(predicate)
+    {}
+
+    inline event_filter::event_filter(unsigned short event_id, filter_predicate predicate/*=nullptr*/)
+    : provider_filter_event_ids_{ event_id },
```

Comment:
should this be `({ event_id })` instead of `{ event_id }`?

Reply:
You forgot a possible `{{ event_id }}` (this language is a disaster).

```
template <int x>
struct Tag { };

class Foo
{
	std::vector<short> vs_;

public:								      // Foo f(5)
	Foo(short s, Tag<1>) : vs_(s) { }     // make a vector with 5 entries
	Foo(short s, Tag<2>) : vs_{ s } { }	  // make a vector with 1 entry, value 5
	Foo(short s, Tag<3>) : vs_({ s }) { } // make a vector with 1 entry, value 5
	Foo(short s, Tag<4>) : vs_{{ s }} { } // make a vector with 1 entry, value 5

	auto count() const { return vs_.size(); }
	auto first() const { return *vs_.begin(); }
};

int main() noexcept
{
	Foo f1(5, Tag<1>{});
	Foo f2(5, Tag<2>{});
	Foo f3(5, Tag<3>{});
	Foo f4(5, Tag<4>{});
	std::cout << f1.count() << ", " << f1.first() << std::endl;
	std::cout << f2.count() << ", " << f2.first() << std::endl;
	std::cout << f3.count() << ", " << f3.first() << std::endl;
	std::cout << f4.count() << ", " << f4.first() << std::endl;
}
```

Reply:
Short answer, it's fine.